autogpts/autogpt/autogpt/agents/agent_member.py

```diff
@@ -0,0 +1,316 @@
+        super().__init__(settings, llm_provider, command_registry, file_storage, config)
+
+        self.role = role
+        self.initial_prompt = initial_prompt
+        self.boss = boss
+        self.recruiter = recruiter
+        self.tasks = []
+        self.members = []
+        self.create_agent = create_agent
+        self.prompt_strategy = DivideAndConquerAgentPromptStrategy(
+            configuration=settings.prompt_config,
+            logger=logger,
+        )
+
+    def build_prompt(
+        self,
+        scratchpad: PromptScratchpad,
+        tasks: list["AgentTask"],
+        extra_commands: Optional[list[CompletionModelFunction]] = None,
+        extra_messages: Optional[list[ChatMessage]] = None,
+        **extras,
+    ) -> ChatPrompt:
+        """Constructs a prompt using `self.prompt_strategy`.
+
+        Params:
+            scratchpad: An object for plugins to write additional prompt elements to.
+                (E.g. commands, constraints, best practices)
+            extra_commands: Additional commands that the agent has access to.
+            extra_messages: Additional messages to include in the prompt.
+        """
+        if not extra_commands:
+            extra_commands = []
+        if not extra_messages:
+            extra_messages = []
+
+        # Apply additions from plugins
+        for plugin in self.config.plugins:
+            if not plugin.can_handle_post_prompt():
+                continue
+            plugin.post_prompt(scratchpad)
+        ai_directives = self.directives.copy(deep=True)
+        ai_directives.resources += scratchpad.resources
+        ai_directives.constraints += scratchpad.constraints
+        ai_directives.best_practices += scratchpad.best_practices
+        extra_commands += list(scratchpad.commands.values())
+
+        prompt = self.prompt_strategy.build_prompt(
+            include_os_info=True,
+            tasks=tasks,
+            agent_member=self,
+            ai_profile=self.ai_profile,
+            ai_directives=ai_directives,
+            commands=get_openai_command_specs(
+                self.command_registry.list_available_commands(self)
+            )
+            + extra_commands,
+            event_history=self.event_history,
+            max_prompt_tokens=self.send_token_limit,
+            count_tokens=lambda x: self.llm_provider.count_tokens(x, self.llm.name),
+            count_message_tokens=lambda x: self.llm_provider.count_message_tokens(
+                x, self.llm.name
+            ),
+            extra_messages=extra_messages,
+            **extras,
+        )
+
+        return prompt
+
+    async def execute_commands(
+        self, commands: list["Command"]
+    ) -> list[CommandActionResult]:
+        # self.log_cycle_handler.log_cycle(
+        #     self.ai_profile.ai_name,
+        #     self.created_at,
+        #     self.config.cycle_count,
+        #     assistant_reply_dict,
+        #     NEXT_ACTION_FILE_NAME,
+        # )
```

Comment:
imo you can remove all `log_cycle` calls, I'll change/remove it in the coming weeks anyway. It's just for debugging.